### ISSUE TITLE
Migrate get evaluations query to Drizzle

### ISSUE THREAD

**Issue by dinmukhamedm** [lmnr-ai/lmnr#81]
Migrate get evaluations, get evaluation, get evaluation datapoint queries from app-server (`routes/evaluations.rs` and `db/evaluations.rs`) directly to calling drizzle from the next routes in `/api/projects/[projectId]/evaluations`.

Get distribution and get score stats are out of scope for now as they involve clickhouse

**Comment by haikalvidya** [comment]
has anyone already started working on this? I also reviewed the source code and noticed that "get evaluations" seems to be implemented - could you confirm if that’s correct?

<img width="822" alt="Screenshot 2024-11-03 at 20 47 20" src="https://github.com/user-attachments/assets/8860f0cd-b019-4de7-ba68-f1ebbeae5159">

**Comment by dinmukhamedm** [comment]
Hey @haikalvidya, thanks for your interest! No one currently is working on this, so feel free to pick it up. Yes, it is correct that plain get evaluations is migrated already, but there is a few more things left to do, namely:
1. get evaluations with the same group id as current evaluation. This query needs refactoring and renaming.
1. get single evaluation. This query pulls both the evaluation metadata and the evaluation results preview.
1. get a single evaluation datapoint.
1. delete evaluation

#### 1. Get evaluations in group.

This is done in the [evaluation page component](https://github.com/lmnr-ai/lmnr/blob/main/frontend/app/project/%5BprojectId%5D/evaluations/%5BevaluationId%5D/page.tsx), where we fetch the evaluation info (with results) and the evaluations within the same group in parallel. This optimization makes not much sense, and will make even less sense once this is done directly from the frontend. The query needs to be done sequentially after getting the evaluation and take in just the group id.

#### 2. Get a single evaluation.

This is done from within the same page, and should probably be split into two queries. See [get_evaluation](https://github.com/lmnr-ai/lmnr/blob/main/app-server/src/db/evaluations.rs#L72-L88) and [get_evaluation_results](https://github.com/lmnr-ai/lmnr/blob/main/app-server/src/db/evaluations.rs#L200-L231) in the app-server.

#### 3. Get a single evaluation

From a brief look it seems unused, we need to confirm this and remove if so.

#### 4. Delete evaluation

This is pretty self-explanatory and should be straightforward.

**Comment by dinmukhamedm** [comment]
Thanks @haikalvidya 
1. I'd probably keep the two API's like now, but fetch the data sequentially in the page. Basically the only change to the queries will be that the second query will take evaluation group id, which we can know only once we've fetched the evaluation itself. In the current state, the second query takes evaluation id and makes an SQL subquery, which was only done in order to run the two in parallel. We do not need that.
2. Yes, that's correct. These two can actually run in parallel, like they are now in the backend: https://github.com/lmnr-ai/lmnr/blob/main/app-server/src/routes/evaluations.rs#L75-L89
3. Yes, and the folder it is in, because it will become empty. Also the endpoint in the backend. That's [here](https://github.com/lmnr-ai/lmnr/blob/main/app-server/src/routes/evaluations.rs#L99-L109) and [here](https://github.com/lmnr-ai/lmnr/blob/main/app-server/src/main.rs#L496), and perhaps [this function](https://github.com/lmnr-ai/lmnr/blob/main/app-server/src/db/evaluations.rs#L241-L263) too, I don't think it is used elsewhere.
4. Ah, good catch. No, no need then, you could remove it, similar to the point 3 above.

Thanks again!

**Comment by haikalvidya** [comment]
already created draft PR, but I haven't tested it yet, and there are some files that have not been deleted as you suggested. Ill work on that and testing it before set the PR to "ready to review"

**Comment by dinmukhamedm** [comment]
Closed by #160
